mmn/node_information.go

```diff
@@ -0,0 +1,125 @@
+package mmn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/mysteriumnetwork/node/metadata"
+)
+
+type NodeInformation struct {
+	MACAddress  string `json:"mac_address"`
+	LocalIP     string `json:"local_ip"`
+	OS          string `json:"os"`
+	Arch        string `json:"arch"`
+	NodeVersion string `json:"node_version"`
+	Identity    string `json:"identity"`
+}
+
+func GetNodeInformation() (*NodeInformation, error) {
+	var mac, ip string
+	ip, err := getLocalNetworkIP()
+
+	if err == nil {
+		mac, err = getMACAddress(ip)
+		if err != nil {
+			mac = ""
```

Comment:
we should not continue if theres an error. We should be returning immediately and caller should not interpret returned data if theres an error.